### PR TITLE
Made node_id mandatory when fetching messages from history

### LINKED ISSUE #2670: Use `node_id` when fetching messages from history
Update `golem.network.history.get()` and `golem.task.tasksession.get_task_message()` to use mandatory `node_id` parameter.

### PR DESCRIPTION
**Pull request by kmazurek**
fixes: #2670

### CODE REVIEW

File: golem/network/history.py [hunk 1]
```diff
@@ -330,26 +329,20 @@ def add(msg: message.base.Message,
...
         subtask_id: str,
-        node_id: Optional[str] = None,
+        node_id: str,
         task_id: Optional[str] = None) \
         -> Optional[message.base.Message]:
-    #  FIXME: Use node_id in queries
-    #         https://github.com/golemfactory/golem/issues/2670
-    if not (node_id or task_id):
+    if not (node_id and subtask_id):
         raise RuntimeError("Please use node_id, subtask_id pair")
```
Comment (etam) [inline golem/network/history.py]: Is this check still relevant?

Comment (kmazurek) [reply golem/network/history.py]: You're right, it's no longer needed here, let me remove it

File: golem/network/history.py [hunk 2]
```diff
@@ -330,26 +329,17 @@ def add(msg: message.base.Message,
...
         task_id: Optional[str] = None) \
         -> Optional[message.base.Message]:
-    #  FIXME: Use node_id in queries
-    #         https://github.com/golemfactory/golem/issues/2670
-    if not (node_id or task_id):
-        raise RuntimeError("Please use node_id, subtask_id pair")
-
-    query = {}
-    query['subtask'] = subtask_id
+    query = dict()
```
Comment (shadeofblue) [inline golem/network/history.py:335]: https://stackoverflow.com/questions/5790860/and-vs-list-and-dict-which-is-better#5790954a

... just sayin'

**Review by shadeofblue — APPROVED** [review]
lgtm, minor comment added...

File: golem/network/history.py (hunk 2 above)
Comment (etam) [reply golem/network/history.py:335]: :open_mouth:

Comment (kmazurek) [reply golem/network/history.py:335]: Okay, interesting - didn't realise there's a difference!

**Review by etam — APPROVED** [review]
ok

**Pull request merged**